integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/statistics/PrometheusStatisticsTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.esb.statistics;
+
+import org.apache.http.HttpResponse;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.wso2.carbon.automation.engine.frameworkutils.FrameworkPathUtil;
+import org.wso2.carbon.automation.extensions.servers.httpserver.SimpleHttpClient;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.carbon.utils.CarbonUtils;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.xml.sax.SAXException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+public class PrometheusStatisticsTest extends ESBIntegrationTest {
+
+    private ServerConfigurationManager scm;
+    private SimpleHttpClient client;
+    private Map<String, String> headers;
+    private int port = 9091;
+    private boolean isPublishingEnabled = false;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+
+        super.init();
+        client = new SimpleHttpClient();
+        headers = new HashMap<>();
+        changeESBConfig();
+    }
+
+    @Test(groups = {"wso2.esb"}, description = "Test if metric data are exposed, if it is enabled in config")
+    public void testPrometheusPublisher() throws Exception {
+
+        headers.put("Accept", "application/json");
+        readConfig();
+        if (isPublishingEnabled) {
```

Comment:
since the purpose of the test case is to test publishing to Prometheus why we have this if condition. Do you what to enable/disable these tests depending on carbon.xml of the testing pack?

Reply:
The test case tests if metric data are exposed if it is enabled in config.